src/java.base/share/classes/sun/net/www/ParseUtil.java

```diff
@@ -200,8 +200,8 @@ public static String decode(String s) {
                 assert (n - i >= 2);
                 try {
                     bb.put(unescape(s, i));
-                } catch (NumberFormatException e) {
-                    throw new IllegalArgumentException();
+                } catch (NumberFormatException | IndexOutOfBoundsException e) {
+                    throw new IllegalArgumentException("Malformed escape pair: " + s);
```

Comment:
I would suggest turning the assert at line 200 into a proper bound check and throw IllegalArgumentException if `n - i >= 2`

Reply:
I meant if `n - i` is **not** `>= 2` of course.